run-tests.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import argparse
+import subprocess
+
+BASEDIR = os.path.abspath(os.path.dirname(__file__))
+
+_formula = "ckan"
+_images = ["centos6"]
+
+
+def get_tag(image, salt=False):
+    tag = "{0}-formula:{1}".format(_formula, image)
+    if salt:
+        return tag + "_salted"
+    else:
+        return tag
+
+
+def image_exists(image):
+    # docker image accept tags only in version >= 1.8...
+    # so workaround with docker history
+    rc = subprocess.call("docker history -q {0} > /dev/null".format(image), shell=True)
+    if rc == 0:
+        return True
+    elif rc == 1:
+        return False
+    else:
+        raise RuntimeError("Cannot test if image exists")
+
+
+def _build(image, salt=False):
+    dockerfile = "test/{0}.Dockerfile".format(image)
+    tag = get_tag(image, salt)
+    if salt:
+        dockerfile_content = open(dockerfile, "rb").read()
+        dockerfile_content += (
+            b"\n"
+            b"ADD test/minion.conf /etc/salt/minion.d/minion.conf\n"
+            b"ADD test/salt /srv/salt\n"
+            b"ADD test/pillar /srv/pillar\n"
+            b"ADD . /srv/formula\n"
```

Comment:
Philippe Pepiot a écrit :
> I think you should explicitly ADD wanted directories here (ckan,
> _states). If not any change to the current directory (temp file, .git
> and other unrelated files) will disable the use of the docker cache and
> trigger (possibly unwanted) rebuilds when using |./run-tests build|.
>


Done in 05507c3bf991ae06a666b0b8167c149d58e902c7.